direct-query-core/src/main/java/org/opensearch/sql/directquery/DirectQueryExecutorServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.directquery;
+
+import java.io.IOException;
+import java.util.UUID;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.sql.datasource.client.DataSourceClient;
+import org.opensearch.sql.datasource.client.DataSourceClientFactory;
+import org.opensearch.sql.datasource.client.exceptions.DataSourceClientException;
+import org.opensearch.sql.datasource.query.QueryHandlerRegistry;
+import org.opensearch.sql.directquery.rest.model.ExecuteDirectQueryRequest;
+import org.opensearch.sql.directquery.rest.model.ExecuteDirectQueryResponse;
+import org.opensearch.sql.directquery.rest.model.GetDirectQueryResourcesRequest;
+import org.opensearch.sql.directquery.rest.model.GetDirectQueryResourcesResponse;
+import org.opensearch.sql.directquery.rest.model.WriteDirectQueryResourcesRequest;
+import org.opensearch.sql.directquery.rest.model.WriteDirectQueryResourcesResponse;
+
+/*
+ * @opensearch.experimental
+ */
+public class DirectQueryExecutorServiceImpl implements DirectQueryExecutorService {
+
+  private final DataSourceClientFactory dataSourceClientFactory;
+  private final QueryHandlerRegistry queryHandlerRegistry;
+
+  @Inject
+  public DirectQueryExecutorServiceImpl(
+      DataSourceClientFactory dataSourceClientFactory, QueryHandlerRegistry queryHandlerRegistry) {
+    this.dataSourceClientFactory = dataSourceClientFactory;
+    this.queryHandlerRegistry = queryHandlerRegistry;
+  }
+
+  @Override
+  public ExecuteDirectQueryResponse executeDirectQuery(ExecuteDirectQueryRequest request) {
+    // TODO: Replace with the data source query id.
+    String queryId = UUID.randomUUID().toString();
```

Comment:
direct query (non async query) shouldn't have a query id?

Reply:
I was thinking of keeping this for now for the purpose of tracing the call to debug errors.